Realm/Tests/MigrationTests.mm

```diff
@@ -243,6 +243,34 @@ - (void)testGetSchemaVersion {
     XCTAssertEqual(1U, [RLMRealm schemaVersionAtURL:config.fileURL encryptionKey:nil error:nil]);
 }
 
+- (void)testGetSchemaVersionOnReadOnlyRealm {
+    NSFileManager *fileManager = [NSFileManager defaultManager];
+    NSURL *directoryURL = [RLMDefaultRealmURL() URLByDeletingLastPathComponent];
+    
+    NSError *error = nil;
+    NSURL *readOnlyDirectoryURL = [directoryURL URLByAppendingPathComponent:@"ReadOnly"];
+    XCTAssertTrue([fileManager createDirectoryAtURL:readOnlyDirectoryURL withIntermediateDirectories:YES attributes:nil error:&error]);
+    XCTAssertNil(error);
+    
+    RLMRealmConfiguration *config = [RLMRealmConfiguration defaultConfiguration];
+    config.fileURL = [readOnlyDirectoryURL URLByAppendingPathComponent:@"default.realm"];
+    @autoreleasepool { [RLMRealm realmWithURL:config.fileURL]; }
+    XCTAssertEqual(0U, [RLMRealm schemaVersionAtURL:config.fileURL encryptionKey:nil error:nil]);
+    
+    error = nil;
+    XCTAssertTrue([fileManager setAttributes:@{NSFilePosixPermissions: @0x100} ofItemAtPath:config.fileURL.path error:&error]);
+    XCTAssertTrue([fileManager setAttributes:@{NSFilePosixPermissions: @0x100} ofItemAtPath:readOnlyDirectoryURL.path error:&error]);
```

Comment:
This leaves a directory named ReadOnly that takes special work to remove (i.e., `rm -rf` chokes on it). It'd be worth having the test case clean up after itself as best it can.